src/main/java/com/cburch/logisim/std/ttl/Ttl74381.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * This file is part of logisim-evolution.
+ *
+ * Logisim-evolution is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * Logisim-evolution is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with logisim-evolution. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Original code by Marcin Orlowski (http://MarcinOrlowski.com), 2021.
+ */
+
+package com.cburch.logisim.std.ttl;
+
+import com.cburch.logisim.data.Value;
+import com.cburch.logisim.instance.InstancePainter;
+import com.cburch.logisim.instance.InstanceState;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * TTL 74x381 arithmetic logic unit
+ * Model based on <a href="http://bitsavers.org/components/ti/_dataBooks/1976_TI_The_TTL_Data_Book_2ed/07.pdf">74LS181 datasheet, p 7-484 to 7-486</a>.
+ */
+public class Ttl74381 extends AbstractTtlGate {
+  /**
+   * Unique identifier of the tool, used as reference in project files.
+   * Do NOT change as it will prevent project files from loading.
+   * Identifier value MUST be unique string among all tools.
+   */
+  public static final String _ID = "74381";
+
+  public static final int DELAY = 1;
+
+  // IC pin indices as specified in the datasheet
+
+  // Inputs
+  public static final byte A0 = 3;
+  public static final byte A1 = 1;
+  public static final byte A2 = 19;
+  public static final byte A3 = 17;
+
+  public static final byte B0 = 4;
+  public static final byte B1 = 2;
+  public static final byte B2 = 18;
+  public static final byte B3 = 16;
+
+  public static final byte F0 = 8;
+  public static final byte F1 = 9;
+  public static final byte F2 = 11;
+  public static final byte F3 = 12;
+
+  public static final byte S0 = 5;
+  public static final byte S1 = 6;
+  public static final byte S2 = 7;
+
+  public static final byte Ci = 15;
+
+  // Outputs
+  public static final byte P = 14;
+  public static final byte G = 13;
+
+  // Power supply
+  public static final byte GND = 10;
+  public static final byte VCC = 20;
+
+  private InstanceState _state;

```

Comment:
Please remove `_` prefix. We do not use any prefix for globals. If you need to mark the property in any special way, just name it as such.

Reply:
Done